dietza/whats-cookin
# Classes & Testing

# Pull Request Template:

#### What's this PR do?
Completes building out the User class in `User.js`, adding and passing corresponding tests in `User-test.js`.
Includes moving methods previously defined by other classes to the User class.

Begins development of structure and style in `index.html` and `styles.css`

#### Where should the reviewer start?
Start in the `User.js` class file.
Consult project spec / rubric for adherence:
Overview the dependencies and sample data sets at the top of the `User-test.js` testing file.

#### How should this be manually tested?
The reviewer should manually test this by using the previously installed mocha and chai testing frameworks to run `npm test`.
From a local copy of the repository, run `npm test ../test/Recipe-test.js`

To view beginning stages of the page, run `open index.html` to view in browser.

#### Any background context you want to provide?
The reviewer should be aware that the datasets provided (sampleIngredients, sampleUsers, and sampleRecipes) on lines 6- 754 of the `User-test.js` testing suite are shallow copies of the full datasets available in respective files within the data directory.

Additionally, there are inconsistencies present in the commits (where development began on `index.html` and `styles.css`) before continued development for those files was appropriately moved to a different branch.
